bigtable/spark/README.md

```diff
@@ -0,0 +1,458 @@
+
+### Configure Environment
+
+```
+BIGTABLE_SPARK_PROJECT_ID=your-project-id
+BIGTABLE_SPARK_INSTANCE_ID=your-bigtable-instance-id
+BIGTABLE_SPARK_DATAPROC_CLUSTER=spark-cluster
+BIGTABLE_SPARK_REGION=europe-west4
+BIGTABLE_SPARK_JAR=target/scala-2.11/bigtable-spark-samples-assembly-0.1.jar
+BIGTABLE_SPARK_CLASS=example.DataFrameDemo
+BIGTABLE_SPARK_TABLE=DataFrameDemo
+```
+
+**NOTE** `BIGTABLE_SPARK_REGION` should point to your region. Read [Available regions and zones](https://cloud.google.com/compute/docs/regions-zones#available) in the official documentation.
+
+### Authenticate
+
+Authenticate to a Google Cloud Platform API using service or user accounts.
+Learn about [authenticating to a GCP API](https://cloud.google.com/docs/authentication/) in the Google Cloud documentation.
+
+**NOTE**: In most situations, we recommend [authenticating as a service account](https://cloud.google.com/docs/authentication/production) to a Google Cloud Platform (GCP) API.
+
+```
+GOOGLE_APPLICATION_CREDENTIALS=/your/service/account.json
+```
+
+### Create Google Cloud Dataproc Cluster
+
+```
+BIGTABLE_SPARK_DATAPROC_CLUSTER=spark-cluster
+BIGTABLE_SPARK_REGION=your-region
+BIGTABLE_SPARK_PROJECT_ID=your-project-id
+
+gcloud dataproc clusters create $BIGTABLE_SPARK_DATAPROC_CLUSTER \
+  --region=$BIGTABLE_SPARK_REGION \
+  --project=$BIGTABLE_SPARK_PROJECT_ID
+```
+
+### Configure Cloud Bigtable
+
+```
+BIGTABLE_SPARK_INSTANCE_ID=your-instance-id
+BIGTABLE_SPARK_CLUSTER_ID=your-cluster-id
+BIGTABLE_SPARK_CLUSTER_ZONE=your-cluster-zone
+
+gcloud bigtable instances create $BIGTABLE_SPARK_INSTANCE_ID \
+    --cluster=$BIGTABLE_SPARK_CLUSTER_ID \
+    --cluster-zone=$BIGTABLE_SPARK_CLUSTER_ZONE \
+    --display-name=$BIGTABLE_SPARK_INSTANCE_ID
+```
+
+```
+BIGTABLE_SPARK_PROJECT_ID=bigtable-spark-connector
+BIGTABLE_SPARK_DataFrameDemo_TABLE=DataFrameDemo
+cbt \
+  -project=$BIGTABLE_SPARK_PROJECT_ID \
+  -instance=$BIGTABLE_SPARK_INSTANCE_ID \
+  createtable $BIGTABLE_SPARK_DataFrameDemo_TABLE
+```
+
+```
+cbt \
+  -project=$BIGTABLE_SPARK_PROJECT_ID \
+  -instance=$BIGTABLE_SPARK_INSTANCE_ID \
+  ls
+```
+
+```
+cbt \
+  -project=$BIGTABLE_SPARK_PROJECT_ID \
+  -instance=$BIGTABLE_SPARK_INSTANCE_ID \
+  createfamily $BIGTABLE_SPARK_DataFrameDemo_TABLE rowkey
+
+cbt \
+  -project=$BIGTABLE_SPARK_PROJECT_ID \
+  -instance=$BIGTABLE_SPARK_INSTANCE_ID \
+  createfamily $BIGTABLE_SPARK_DataFrameDemo_TABLE cf1
+
+cbt \
+  -project=$BIGTABLE_SPARK_PROJECT_ID \
+  -instance=$BIGTABLE_SPARK_INSTANCE_ID \
+  createfamily $BIGTABLE_SPARK_DataFrameDemo_TABLE cf2
+
+cbt \
+  -project=$BIGTABLE_SPARK_PROJECT_ID \
+  -instance=$BIGTABLE_SPARK_INSTANCE_ID \
+  createfamily $BIGTABLE_SPARK_DataFrameDemo_TABLE cf3
+```
+
+### Submit Wordcount Job
+
+Submit the Wordcount job to a Cloud Dataproc instance.
+
+```
+gcloud dataproc jobs submit spark \
+  --cluster=$BIGTABLE_SPARK_CLUSTER \
```

Comment:
this is not left as an instruction on what to set as - maybe switch this to `BIGTABLE_SPARK_CLUSTER_ID`, which is already defined elsewhere?